vagrant/setup.sh

```diff
@@ -101,6 +105,15 @@ fi
 echo '-----> Running pip install'
 pip install --require-hashes -r requirements/common.txt -r requirements/dev.txt | sed -e '/^Requirement already satisfied:/d'
 
+if [[ $(geckodriver --version) != *$GECKODRIVER_VERSION* ]]; then
```

Comment:
By default `$(...)` only captures stdout not stderr, so for a fresh provision this step currently outputs:
`./vagrant/setup.sh: line 108: geckodriver: command not found`

To prevent this, I'd redirect stderr to stdout like for the installation of pip above. I'd also quote the variables, so something like:
`if [[ "$(geckodriver --version 2>&1)" != *"${GECKODRIVER_VERSION}"* ]]; then`

I'm looking forward to using Docker where we won't have to do this version checking boilerplate and also don't have the duplication between this script and `travis-setup.sh` :-)
(It didn't seem worth trying to overcomplicate bug 1414094 by making it also consolidate the duplication, given both scripts will be replaced by the Docker bootstrap process)

Reply:
I did try this first, and wasn't able to get it to work, but I will give it another go.